### PR TITLE
Performance of vec_fmt_*() by avoiding checking of compatibility + coercion to tibble

### PR DESCRIPTION
**Pull request by olivroy**
# Summary

Hopefully, my last one. 

My example is down to about 4 seconds rendering time.

The main idea is to avoid dplyr or coercion to tibble and also avoid to check for the compatibility of the formatter unnecessarily. 

I realized `cols_add()` has some issues, so I fixed up the two corner cases `.before = 0` and `.after = last_col()` + added tests.

Also eliminated some tidyselect deprecation warnings

# Related GitHub Issues and PRs

- Ref: #1888 , #1891

### CODE REVIEW

File: tests/testthat/_snaps/cols_add.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+    Code
+      cols_add(tbl, x = 1, .after = 2, .before = 3)
+    Condition
+      Error in `cols_add()`:
+      ! Expressions cannot be given to both `.before` and `.after`.
+    Code
+      cols_add(tbl, x = 1, .after = 15)
+    Condition
+      Error in `vars_select_eval()`:
```
Comment (olivroy) [inline tests/testthat/_snaps/cols_add.md:11]: Will be fixed by https://github.com/r-lib/tidyselect/pull/352 when merged

File: R/utils_render_common.R [hunk 2]
```diff
@@ -108,13 +108,17 @@ render_formats <- function(data, context) {
...
-          compat = compat
+        skip_compat_check ||
+        (
+          col %in% colnames(data_tbl) &&
+          is_compatible_formatter(
+            table = data_tbl,
+            column = col,
+            rows = rows,
+            compat = compat
+          )
```
Comment (olivroy) [inline R/utils_render_common.R:121]: I added this to skip this check for vec_fmt_*

File: R/utils.R [hunk 3]
```diff
@@ -226,9 +226,9 @@ tf_formats <- function() {
 
   dplyr::tribble(
```
Comment (olivroy) [inline R/utils.R:227]: Before, it seemed that `tf_formats_icons()` included NAs, not sure what the impact of all this is tbh.

Now, it is:

```
tf_formats_icons()
[1] "c(\"✔\", \"✘\")" "c(\"●\", \"⭘\")" "c(\"■\", \"□\")" "c(\"◆\", \"◇\")" "c(\"↑\", \"↓\")" "c(\"▲\", \"▼\")"
[7] "c(\"▶\", \"◀\")"
```

which seems to make sense?

Comment (rich-iannone) [reply R/utils.R:227]: Thanks for adding `omit_na()`! Yeah, this return value is correct now but AFAICT nothing in the package uses `tf_formats_icons()` (so I guess no impact).

**Review by rich-iannone — APPROVED** [review]
LGTM!
